src/autoqasm/operators/typecast.py

```diff
@@ -0,0 +1,36 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+
+"""Operators for int cast statements."""
+from __future__ import annotations
+
+from typing import Any
+
+from autoqasm import types as aq_types
+
+
+def int_typecast(argument_: Any, *args, **kwargs) -> aq_types.IntVar | int:
+    """Operator declares the `oq` variable, or sets variable's value if it's
+    already declared.
+
+    Args:
+        argument_ (Any): object to be converted into an int.
+
+    Returns:
+        IntVar | int : IntVar object if argument is QASM type, else int.
+    """
+    if aq_types.is_qasm_type(argument_):
+        return aq_types.IntVar(argument_)
```

Comment:
Super clear! Thank you so much.

I wanted to double-check one last thing. If we consider the test:

```
@aq.main(num_qubits=2)
        def main():
            test = int(measure([0, 1]))
```
I am now getting the following instructions:
```
int[32] test;
qubit[2] __qubits__;
bit[2] __bit_0__ = "00";
__bit_0__[0] = measure __qubits__[0];
__bit_0__[1] = measure __qubits__[1];
int[32] __int_1__;
__int_1__ = __bit_0__;
test = __int_1__;
```
I am particularly interested in the last 3 lines. Would you have expected something like `test = __bit_0__` (given that `test` was already declared to be an `int` in the first line) or is declaring `__int_1__` and setting it equal to `__bit_0__` something expected due to the conversion?

Reply:
Yes, this is completely expected because of the way the conversion is happening. You're creating an IntVar (`__int_1__`) from the typecast, and then the assignment is creating another IntVar (`test`).

In the future we could certainly look for ways to optimize the generated IR, but right now we don't do that.